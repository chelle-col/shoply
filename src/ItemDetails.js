import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addToCart, changeAmount } from './CartActions'

const ItemDetails = ({ itemArray }) => {
    const [ amount, setAmount ] = useState(1)
    const [ id, item ] = itemArray;
    const dispatch = useDispatch();
    const cart = useSelector(st => st.cart)

    // const changeAmount = ( itemId, amount ) => dispatch({ type: "CHANGE_AMOUNT", amount, id: itemId});
    
    const handleClick = (e) => {
        setAmount( amount => amount + 1 );
        const itemId = e.target.id;
        const item = cart.filter( i => i.id === itemId);
        if( item.length === 0 ){
            dispatch( addToCart( itemId ));
        }
    }

    useEffect( ()=> {
        const item = cart.filter( i => i.id === id);
        if( item.length > 0 ){
            dispatch(changeAmount( id, amount ));
        }
        console.log(cart[0])
    }, [ amount ])

    return (
        <div>
            <h3>{item.name}</h3>
            <p>{item.description}</p>
            <img style={{width:'250px'}} src={item.image_url} alt='' />
            <div>
                <button onClick={handleClick} id={id}>${item.price}</button>
            </div>
        </div>
    )
}

export default ItemDetails;