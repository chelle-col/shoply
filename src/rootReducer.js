let reduxStore = require( "./shoppingData.json" );

const INITIAL_STATE = {
    mart: reduxStore,
    cart: []
}

function rootReducer(state = INITIAL_STATE, action) {
    switch (action.type) {
      case "ADD":
        return { ...state, 
            cart: 
                [...state.cart, 
                    { id: action.id, 
                        amount: action.amount} 
                ] 
            };

      case "CHANGE_AMOUNT":
          return{...state,
            cart: state.cart.map( (item) => {
                if( item.id !== action.id ){
                    return item;
                }
                return {
                    ...item,
                    amount: action.amount
                };
            })}
  
      case "REMOVE":
        return { ...state, 
            cart: state.cart.filter( (item)=> item.id !== action.id)
        }
  
      default:
        return state;
    }
  }
  
  export default rootReducer;