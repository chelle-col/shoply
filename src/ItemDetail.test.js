import ItemDetails from './ItemDetails';
import rootReducer from './rootReducer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from "@testing-library/react";

const store = createStore( rootReducer );
const testItem = {
    "1":
    {
    "name": "testItem",
    "price": 100.00,
    "description": "tesing",
    "image_url": "https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
    }
};



it("Should Render", function () {
    render(
        <Provider store={store}>
            <ItemDetails itemArray={Object.entries(testItem)[0]}/>
        </Provider>
        );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}>
            <ItemDetails itemArray={Object.entries(testItem)[0]}/>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});