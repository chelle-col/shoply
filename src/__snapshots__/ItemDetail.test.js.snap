// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      testItem
    </h3>
    <p>
      tesing
    </p>
    <img
      alt=""
      src="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
      style="width: 250px;"
    />
    <div>
      <button
        id="1"
      >
        $100
      </button>
    </div>
  </div>
</DocumentFragment>
`;
